DragGAN/draggan_onnx_demo.py

```diff
@@ -0,0 +1,342 @@
+    
+    onnx_output_folder = os.path.join(current_script_folder, "onnx", model_name)
+    if (not os.path.isdir(onnx_output_folder)):
+        os.makedirs(onnx_output_folder)
+
+    params_path = os.path.join(model_folder, model_name + ".json")
+
+    with open(params_path) as f:
+        params = json.load(f)
+
+    ws_feats = 512
+    feat_list  = params['genFeatureList'] 
+    map_layers = params['mapLayers']
+    gen_layers = params['genLayers']
+
+    # creating the onnx mapper session
+    onnx_mapper_filename = os.path.join(onnx_output_folder, "stylegan_mapper.onnx")
+    execution_providers = ["CUDAExecutionProvider", "CPUExecutionProvider"]
+
+    print(f"Creating ONNX Mapper session for [{onnx_mapper_filename}]")
+    so = onnxruntime.SessionOptions()
+    mapper_session = onnxruntime.InferenceSession(
+        onnx_mapper_filename, so, providers=execution_providers
+    )
+
+
+    # loading the checkpoint data
+    checkpoint_filename = f"{onnx_output_folder}/checkpoint"
+    print(f"Creating ONNX Mapper session for [{checkpoint_filename}]")
+    checkpoint_state = orttraining.CheckpointState.load_checkpoint(checkpoint_filename)
+
+    # creating the training module
+    print(f"Creating ONNX Training Module")
+    model = orttraining.Module(
+        f"{onnx_output_folder}/training_model.onnx",
+        checkpoint_state,
+        f"{onnx_output_folder}/eval_model.onnx",
+        device="cuda",
+    )
+
+    optimizer_filename = f"{onnx_output_folder}/optimizer_model.onnx"
+    print(f"Creating the ONNX Optimizer from [{checkpoint_filename}]")
+    optimizer = orttraining.Optimizer(optimizer_filename, model)
+
+    with open(os.path.join(current_script_folder, "handles.json")) as user_file:
+        parsed_json = json.load(user_file)
+    print(parsed_json)
+
+    model_snapshot = model.get_contiguous_parameters(trainable_only=True)
+
+    seed = 71
+    print(f"running with seed [{seed}]")
+    ws = generate_latent_vector(mapper_session, seed, ws_feats)
+    update_model(model, model_snapshot, ws)
+
+    scale = 1.0
+    handles = [(p[1]*scale, p[0]*scale) for p in parsed_json['points']]
+    targets = [(p[3]*scale, p[2]*scale) for p in parsed_json['points']]
+
+    output_path = os.path.join(current_script_folder, "out", "orttraining1")
+    optimize(model, optimizer, handles, targets, output_path)
+
+    ffmpeg_create_video(output_path)
+
+
+    seed = 85
+    print(f"running with seed [{seed}]")
+    ws = generate_latent_vector(mapper_session, seed, ws_feats)
+    update_model(model, model_snapshot, ws)
+
+    scale = 1.0
+    handles = [(p[1]*scale, p[0]*scale) for p in parsed_json['points']]
+    targets = [(p[3]*scale, p[2]*scale) for p in parsed_json['points']]
+
+    output_path = os.path.join(current_script_folder, "out", "orttraining2")
+    optimize(model, optimizer, handles, targets, output_path)
+
+    ffmpeg_create_video(output_path)
+
+    print("done")
+
```

Comment:
the code should be cleaned with black.

Reply:
not sure what black is. i removed some of the commented old code. if there is an automatic tool that does that i would be happy to apply it.